src/System.Private.CoreLib/shared/System/Threading/Tasks/ConcurrentExclusiveSchedulerPair.cs

```diff
@@ -516,7 +518,7 @@ void IThreadPoolWorkItem.Execute()
         private sealed class ConcurrentExclusiveTaskScheduler : TaskScheduler
         {
             /// <summary>Cached delegate for invoking TryExecuteTaskShim.</summary>
-            private static readonly Func<object, bool> s_tryExecuteTaskShim = new Func<object, bool>(TryExecuteTaskShim);
+            private static readonly Func<object?, bool> s_tryExecuteTaskShim = new Func<object?, bool>(TryExecuteTaskShim);
```

Comment:
Just to understand, why is `object` nullable in these context if `TryExecuteTaskShim` doesn't accept a nullable object?

Reply:
Hmm, TryExecuteTaskShim should be taking `object?` (all of this predicated on not having attributes that allow us to annotate the non-nullness appropriately).  I'll fix that and also figure out if this is a compiler bug or something else is going on.

Reply:
I see, it is because we use it to construct a Task that takes this type: https://github.com/dotnet/coreclr/pull/23691/files#diff-fc19852be9d00ad770ca78f35eb6e6f3R213

Reply:
>Hmm, TryExecuteTaskShim should be taking object? (all of this predicated on not having attributes that allow us to annotate the non-nullness appropriately)

Could it be hitting the issue that we found earlier this week related to static fields? Where it doesn't seem to be doing nullable analysis on them? It doesn't warn on a mismatch of type when assigning them, nor uninitialized static fields that are non-nullable.

Reply:
https://github.com/dotnet/roslyn/issues/34731